src/bin/ion/commands/generate/templates/typescript/interface.templ

```diff
@@ -0,0 +1,162 @@
+{% import "util_macros.templ" as macros %}
+
+import * as ion from 'ion-js';
+import { IonSerializable, StructType } from './ion_generated_code';
+
+{% if model.code_gen_type and model.code_gen_type.doc_comment %}
+/**
+ * {{ model.code_gen_type.doc_comment }}
+ */
+{% endif %}
+{% if model.annotations %}
```

Comment:
The preview version of code generation does not support annotations at this point. I think it would be best if we can add this support separately so that its consistent with other languages(i.e. Java and Rust).